### ISSUE TITLE
Does tf.math.equal supports tf.sparse.SparseTensor?

### ISSUE THREAD

**Issue by HongtaoYang** [tensorflow/tensorflow#50189]
## URL with the issue:
https://www.tensorflow.org/api_docs/python/tf/math/equal

## Description of issue (what needs changing):
The doc says `tf.math.equal` supports sparse tensor as inputs, but it is actually not supported at the moment. See the example below. I haven't checked all math operations that claims to support sparse tensors, maybe there are other similar doc errors like this.

I'm using tf `2.6.0-dev20210601`.

### Usage example
```python
import tensorflow as tf

a = tf.sparse.SparseTensor(
    indices=[[0, 0], [0, 1], [1, 2]],
    values=[1, 1, 1],
    dense_shape=[2, 3]
)

b = tf.sparse.SparseTensor(
    indices=[[0, 0], [0, 1], [1, 2]],
    values=[1, 1, 1],
    dense_shape=[2, 3]
)

tf.math.equal(a, b)  # raises ValueError
```

I think making `equal` operation supports sparse tensor is needed.

**Comment by tilakrayal** [comment]
@HongtaoYang ,

We see that the issue [template](https://github.com/tensorflow/tensorflow/issues/new/choose) has not been filled, could you please do so as it helps us analyse the issue.Thanks!

**Comment by Namangarg110** [comment]
Hey I'm new to contributing to open source, can I work on this issue?

**Comment by HongtaoYang** [comment]
Thanks for working on the issue. I'm using this temporary hack as a workaround atm.

```python
import tensorflow as tf
from tensorflow.sparse import SparseTensor


def sparse_tensor_not_equal(a: SparseTensor, b: SparseTensor) -> SparseTensor:
    diff = tf.math.abs(tf.sparse.add(tf.math.negative(a), b))
    diff = tf.cast(diff, bool)

    return diff

def sparse_tensor_equal(a: SparseTensor, b: SparseTensor) -> SparseTensor:
    diff = sparse_tensor_not_equal(a, b)
    is_equal = tf.sparse.map_values(tf.math.logical_not, diff)

    return is_equal


a = tf.sparse.SparseTensor(indices=[[0, 0], [0, 1], [1, 2]], values=[1, 1, 1], dense_shape=[2, 3])
b = tf.sparse.SparseTensor(indices=[[0, 1], [0, 2], [1, 1], [1, 2]], values=[1, 1, 1, 1], dense_shape=[2, 3])
print(sparse_tensor_equal(a, b))
```

**Comment by edloper** [comment]
Looking at the code, it looks to me like `tf.math.equal` and `tf.math.not_equal` do not, and have never, supported sparse tensors. (And they only support IndexedSlices in so far as IndexedSlices can be converted to tensors, so it seems odd to call out support for IndexedSlices.)  It looks like this erroneous documentation was added in e0e1efbe0811aa0913ad8400c532b33c76425427.

It would certainly be possible to extend `tf.math.equal` to support sparse tensors.  If this is done, we should be sure to document the semantics -- namely that `tf.equal(x, y)` iff `tf.equal(tf.sparse.to_dense(x), tf.sparse.to_dense(y))` (so two sparse tensors are considered equal even if one has an explict nonzero where another has an implicit nonzero).  Also, we should document that `x` and `y` are assumed to be in standard lexicographic order (and tf.sparse.reorder needs to be used if they are not).

As for other ops in `math_ops.py` that claim to support `SparseTensors`, I didn't see any similar documentation errors.  In particular, the following claim to work and do work: `tf.math.abs`, `tf.math.negative`, `tf.math.sign`, `tf.cast`.  Also, `my_dense * my_sparse` works (using the multiplication operator), though I don't see that fact documented anywhere.  (In contrast, `tf.math.multiply(my_dense, my_sparse)` does not work).

**Comment by sushreebarsa** [comment]
@HongtaoYang 
Is this still an issue ?
Could you please refer to the [comment](https://github.com/tensorflow/tensorflow/issues/50189#issuecomment-862413890) above and let us know if it helps?
Thanks!

**Comment by HongtaoYang** [comment]
Thanks, the comment helps, thanks @edloper ! I'll close this issue.

**Comment by edloper** [comment]
Fixed the documentation in 011e57d43fbbbce20163ca807f29b21d3f0f6084.
